src/watchmaker/workers/salt.py

```diff
@@ -774,10 +774,22 @@ def _configuration_validation(self):
                     'parameter `git_repo` was not provided.'
                 )
 
+    def _check_salt_version(self):
+        current_salt_version = self.call_process(['salt-call', '--version'])
+        if self.salt_version and self.salt_version in current_salt_version:
+            self.log.info('Salt is already installed.')
+            return True
+        self.log.info('salt version is not installed')
+        return False
+
     def _install_package(self):
+        if os.path.exists(self.salt_call) and self._check_salt_version():
+            return
         if self.install_method.lower() == 'yum':
             self._install_from_yum(self.yum_pkgs)
         elif self.install_method.lower() == 'git':
+
+            self.log.info('Starting Salt installation')
```

Comment:
If you move this up, just above line 788, then it will log this for every salt install method and not just `git`.